objcgen/embedder.cs

```diff
@@ -629,7 +662,8 @@ public int Compile ()
 								mmp.Append ("--debug ");
 							mmp.Append ("-p "); // generate a plist
 							mmp.Append ($"--target-framework {GetTargetFramework ()} ");
-							mmp.Append ($"\"--link_flags=-force_load {Path.GetFullPath (sdk_output_file)}\" ");
+							string extensionFlag = Extension ? "-fapplication-extension" : "";
+							mmp.Append ($"--link_flags={extensionFlag} -force_load {Utils.Quote (Path.GetFullPath (sdk_output_file))} ");
```

Comment:
I'm not sure it's ok. 

It was `"--option value1 value2"`
which, IMO, is better as `--option="value1 value2"`
but now it's `--option=value1 value2` which is not the same